### PR TITLE
Adding IReadOnlyDictionary support (for R3)

### PR DESCRIPTION
**Pull request by ewoutkramer**
Adding IReadOnlyDictionary support to the R3 datatypes and resources.

### CODE REVIEW

**Review by marcovisserFurore — APPROVED** [review]

File: Hl7.Fhir.sln [hunk 1]
```diff
@@ -45,8 +45,6 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Hl7.Fhir.Support.Tests", "c
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "HL7.FhirPath.Tests", "common\src\Hl7.FhirPath.Tests\HL7.FhirPath.Tests.csproj", "{98390A5A-0F65-4275-ADF1-ADF4BD915FA3}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Hl7.Fhir.Support.Poco.Tests", "common\src\Hl7.Fhir.Support.Poco.Tests\Hl7.Fhir.Support.Poco.Tests.csproj", "{88E59825-ADF2-4B46-892F-BCC95B68559F}"
```
Comment (marcovisserFurore) [inline Hl7.Fhir.sln:48]: Why removing this test project? We added this for future unit tests related to pocos.

Comment (marcovisserFurore) [reply Hl7.Fhir.sln:48]: The removal of this project is also affecting the build pipeline: it fails now.

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
See comments

File: Hl7.Fhir.sln (hunk 1 above)
Comment (ewoutkramer) [reply Hl7.Fhir.sln:48]: Shouldn't have happened. I was experimenting in another branch, no idea how this ended up here.

**Review by marcovisserFurore — DISMISSED** [review]

**Pull request merged**